frontend/src/pages/Profile.js

```diff
@@ -28,60 +111,245 @@ import { textColor } from "../assets/jss/colorPalette";
  * @returns {JSX.Element}
  */
 export default function Profile({ user, setUser }) {
+	const [hasPasswordChanged, setHasPasswordChanged] = useState(false);
+	const [passwordErrorText, setPasswordErrorText] = useState("");
 	const changeRoute = useChangeRoute();
 
-	return <CenterFlexBoxMedium>
-		<Grid
-			container
-			spacing={2}
-			alignItems="center"
-			justifyContent="center"
-			direction="column"
-		>
-			<Grid item style={{ width: "100%" }}>
-				<Typography variant="h5" gutterBottom component="div">
-					Personal information
-				</Typography>
-				<TableContainer sx={{ border: 2, borderColor: "gray" }}>
-					<Table>
-						<TableBody>
-							<TableRow>
-								<TableCell style={{ color: textColor, borderBottom: "none" }}>Name</TableCell>
-								<TableCell style={{ color: textColor, borderBottom: "none" }}>{user.name}</TableCell>
-							</TableRow>
-							<TableRow>
-								<TableCell style={{ color: textColor, borderBottom: "none" }}>Email</TableCell>
-								<TableCell style={{ color: textColor, borderBottom: "none" }}>{user.email}</TableCell>
-							</TableRow>
-							<TableRow>
-								<TableCell style={{ color: textColor, borderBottom: "none" }}>Email verified?</TableCell>
-								<TableCell style={{ color: textColor, borderBottom: "none" }}>{user.emailVerification ? "Yes" : <>No <span style={{ textDecorationLine: "underline", cursor: "pointer" }} onClick={() => appwriteApi.sendEmailConfirmation()}>Resent email verification</span></>}</TableCell>
-							</TableRow>
-							<TableRow>
-								<TableCell style={{ color: textColor, borderBottom: "none" }}>Password</TableCell>
-								<TableCell style={{ color: textColor, borderBottom: "none" }}><Button variant="outlined" onClick={() => changeRoute("/user/changePassword")}>Change password</Button></TableCell>
-							</TableRow>
-						</TableBody>
-					</Table>
-				</TableContainer>
-			</Grid>
-			<Grid item style={{ width: "100%" }}>
-				<Typography variant="h5" gutterBottom component="div">
-					Wallet
-				</Typography>
-				<Container sx={{ border: 2, borderColor: "gray" }}>
-					<Wallet setUser={setUser} user={user} sx={{ border: 2, borderColor: "gray" }}/>
-				</Container>
-			</Grid>
-			<Grid item>
-				<Button
-					variant="outlined" style={{ color: "red" }} onClick={() => appwriteApi.deleteCurrentSession().then(() => {
-						setUser(null);
-						changeRoute("/");
-					})}
-				>Logout</Button>
-			</Grid>
-		</Grid>
-
-	</CenterFlexBoxMedium>;
-}
\ No newline at end of file
+	const render = () => (<Box component="form" onSubmit={checkAndSaveProfile} >
+		<EmailStatusBanner isVerified={isEmailVerified}/>
```

Comment:
For perfection, some margin is missing in the mobile version 
![image](https://user-images.githubusercontent.com/58464184/150315058-0b96b33f-5d68-4060-a838-e37b11b81f57.png)

Reply:
> Maybe we should also add a seperat button for changing the password if you want to change the user name but not the password instead of having a button that changes all?

Very good idea, or maybe a "reset" button to reset a profile setting to it's last saved value. I should do more work here.

Reply:
Yes, this is the missing margin that I gave up to get some sleep ;-) .
The reordering logic got more complicated than it should be. I'll try to improve it.